utils/decorators.py

```diff
@@ -0,0 +1,20 @@
+from functools import wraps
+from typing import Callable
+from typing import Tuple
+
+from django.contrib.auth.views import redirect_to_login
+from django.core.handlers.wsgi import WSGIRequest
+
+
+def check_permissions(permissions: Tuple[str], redirect_path: str = "/accounts/login/") -> Callable:
+    def decorator(view_func: Callable) -> Callable:
+        @wraps(view_func)
+        def _wrapped_view(request: WSGIRequest, *args: dict, **kwargs: dict) -> Callable:
+            for permission in permissions:
+                if not getattr(request.user, permission):
```

Comment:
I would base it on database field `user_type`, those properties like `is_admin` might change in the future.

Reply:
Changed. Now we checking `user_type`

Reply:
Looks good, only thing I would like to ask you to add is an assert that passed `allowed_user_types` are correct.

Reply:
OK, In progress